google/cloud/firestore_v1/pipeline_result.py

```diff
@@ -0,0 +1,142 @@
+        execution_time: Timestamp | None = None,
+        create_time: Timestamp | None = None,
+        update_time: Timestamp | None = None,
+    ):
+        """
+        PipelineResult should be returned from `pipeline.execute()`, not constructed manually.
+
+        Args:
+            client: The Firestore client instance.
+            fields_pb: A map of field names to their protobuf Value representations.
+            ref: The DocumentReference or AsyncDocumentReference if this result corresponds to a document.
+            execution_time: The time at which the pipeline execution producing this result occurred.
+            create_time: The creation time of the document, if applicable.
+            update_time: The last update time of the document, if applicable.
+        """
+        self._client = client
+        self._fields_pb = fields_pb
+        self._ref = ref
+        self._execution_time = execution_time
+        self._create_time = create_time
+        self._update_time = update_time
+
+    def __repr__(self):
+        return f"{type(self).__name__}(data={self.data()})"
+
+    @property
+    def ref(self) -> BaseDocumentReference | None:
+        """
+        The `BaseDocumentReference` if this result represents a document, else `None`.
+        """
+        return self._ref
+
+    @property
+    def id(self) -> str | None:
+        """The ID of the document if this result represents a document, else `None`."""
+        return self._ref.id if self._ref else None
+
+    @property
+    def create_time(self) -> Timestamp | None:
+        """The creation time of the document. `None` if not applicable."""
+        return self._create_time
+
+    @property
+    def update_time(self) -> Timestamp | None:
+        """The last update time of the document. `None` if not applicable."""
+        return self._update_time
+
+    @property
+    def execution_time(self) -> Timestamp:
+        """
+        The time at which the pipeline producing this result was executed.
+
+        Raise:
+            ValueError: if not set
+        """
+        if self._execution_time is None:
+            raise ValueError("'execution_time' is expected to exist, but it is None.")
+        return self._execution_time
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Compares this `PipelineResult` to another object for equality.
+
+        Two `PipelineResult` instances are considered equal if their document
+        references (if any) are equal and their underlying field data
+        (protobuf representation) is identical.
+        """
+        if not isinstance(other, PipelineResult):
+            return NotImplemented
+        return (self._ref == other._ref) and (self._fields_pb == other._fields_pb)
+
+    def data(self) -> Any:
+        """
+        Retrieves all fields in the result.
+
+        If a converter was provided to this `PipelineResult`, the result of the
+        converter's `from_firestore` method is returned.
```

Comment:
I can't tell if converters are or will be supported in Python pipelines

Reply:
Good catch, I ended up cutting it from this PR. I'll remove the docstring

Let me know if you think converters would be important to have in Python for launch